File: clients/typescript/examples/collections-insert-one-lexical-vectorize.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get an existing collection
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});
const collection = database.collection("**COLLECTION_NAME**");

// Insert a document into the collection
(async function () {
  const result = await collection.insertOne({
    name: "Jane Doe",
    $vectorize:
      "An athlete who loves biking, hiking, running, and swimming in the outdoors",
    $lexical: "She shares her love of triathlons by coaching kids after school",
  });
})();

File: clients/csharp/examples/collections-insert-one-lexical-vectorize.cs
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Collections;
using DataStax.AstraDB.DataApi.Core;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get an existing collection
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      "**API_ENDPOINT**",
      "**APPLICATION_TOKEN**"
    );
    var collection = database.GetCollection("**COLLECTION_NAME**");

    // Insert a document into the collection
    var document = new Document()
    {
      { "name", "Jane Doe" },
      {
        "$vectorize",
        "An athlete who loves biking, hiking, running, and swimming in the outdoors"
      },
      {
        "$lexical",
        "She shares her love of triathlons by coaching kids after school"
      },
    };
    var result = await collection.InsertOneAsync(document);

    Console.WriteLine(result.InsertedId);
  }
}
